src/java.base/share/classes/com/sun/crypto/provider/ChaCha20Cipher.java

```diff
@@ -226,11 +226,15 @@ protected int engineGetOutputSize(int inputLen) {
     protected AlgorithmParameters engineGetParameters() {
         AlgorithmParameters params = null;
         if (mode == MODE_AEAD) {
+            // In a pre-initialized state or any state without a nonce value
+            // this call should cause a random nonce to be generated, but
+            // not attached to the object.
+            byte[] nonceData = initialized ? nonce : createRandomNonce(null);
```

Comment:
A minor unrelated comment is that createRandomNonce() could be made static.

Reply:
Easy enough to do.